.github/workflows/scripts/slack_notifications/gem_notifier.rb

```diff
@@ -59,13 +59,24 @@ def self.gem_source_code_uri(gem_name)
     info = HTTParty.get("https://rubygems.org/api/v1/gems/#{gem_name}.json")
     raise "Response unsuccessful: #{info}" unless info.success?
 
-    info["source_code_uri"]
+    source_code_uri = info["source_code_uri"]
+    validate_github_url(source_code_uri)
+    source_code_uri
   rescue StandardError => e
     abort("#{e.class}: #{e.message}")
   end
 
+  def self.validate_github_url(url)
+    uri = URI.parse(url)
+    unless uri.host == "github.com" && uri.scheme == "https"
```

Comment:
Since its possible to have a host other than GitHub, perhaps we can just consider the `url.scheme`.
```suggestion
    unless uri.scheme == "https"
```